### PR TITLE
Add scan page

### LINKED ISSUE #3: Take photo
User should have the possibility to take a photo of a document

### LINKED ISSUE #4: Save document
User should have the possibility to save the image and the text

### PR DESCRIPTION
**Pull request by davidwobido**
image

Upload a photo to scan via OCR so you have plain text.
While scanning the image you see the status.
Option to save the text to API.

fixes #3
fixes #4

### CODE REVIEW

File: src/app/Components/AddDocumentForm/AddDocumentForm.module.css [hunk 1]
```diff
@@ -0,0 +1,47 @@
+.form {
+  display: grid;
+  grid-template-rows: 1fr;
+  column-gap: 0.5rem;
+  row-gap: 1rem;
+  place-items: center;
+
+  /* border-radius: 2rem; */
```
Comment (welschoff) [inline src/app/Components/AddDocumentForm/AddDocumentForm.module.css:8]: Delete death Code

File: src/app/Components/AddDocumentForm/AddDocumentForm.module.css [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  background-color: var(--color-neutral-bg);
+  color: var(--color-primary);
+  padding: 0.75rem;
+  border-radius: 1rem;
+}
+
+.form__submit {
+  color: var(--color-neutral-bright);
+  background: var(--color-secondary);
+  border-radius: 24px;
```
Comment (welschoff) [inline src/app/Components/AddDocumentForm/AddDocumentForm.module.css:18]:
```suggestion
  border-radius: 1.5rem;
```

File: src/app/globals.css [hunk 3]
```diff
@@ -55,4 +57,18 @@ p {
 
 body {
   margin: 0;
+  padding: 20px;
+}
+
+button {
+  color: var(--color-neutral-bright);
+  background: var(--color-primary);
+  border-radius: 24px;
```
Comment (welschoff) [inline src/app/globals.css:66]:
```suggestion
  border-radius: 1.5rem;
```

**Review by welschoff — APPROVED** [review]
lgtm :)

File: src/app/Components/AddDocumentForm/AddDocumentForm.module.css [hunk 4]
```diff
@@ -0,0 +1,47 @@
+.form {
+  display: grid;
+  grid-template-rows: 1fr;
```
Comment (lajanzen) [inline src/app/Components/AddDocumentForm/AddDocumentForm.module.css:3]: Do you need this here if you only have one row?

File: src/app/Components/AddDocumentForm/AddDocumentForm.tsx [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+          placeholder="Enter title"
+          value={title}
+          onChange={(event) => setTitle(event.target.value)}
+        />
+        <input
+          type="submit"
+          className={styles.form__submit}
+          disabled={!title || isLoading}
+        />
+      </div>{' '}
```
Comment (lmachens) [inline src/app/Components/AddDocumentForm/AddDocumentForm.tsx:40]:
```suggestion
      </div>
```

File: src/app/Components/AddDocumentForm/AddDocumentForm.tsx [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+      title,
+      text,
+    };
+    await postDocument(document);
+    setTitle('');
+  };
+
+  return (
+    <form onSubmit={handleSubmit} className={styles.form}>
+      <div className="wrapper">
```
Comment (lmachens) [inline src/app/Components/AddDocumentForm/AddDocumentForm.tsx:27]: Are you sure that you need this `div`?

File: src/app/pages/Scan/Scan.module.css [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+  color: var(--color-primary);
+  background: var(--color-neutral-bright);
+  border: 1px solid var(--color-primary);
+}
+
+.button__scan {
+  background-color: var(--color-secondary);
+}
+
+button:disabled {
```
Comment (lmachens) [inline src/app/pages/Scan/Scan.module.css:40]: Be careful using element selectors. This effects all `button` in your project.

File: src/app/pages/Scan/Scan.tsx [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+export default Scan;
+
+// If imageUrl
+//       show button
+//       else if
+//       recognizedProgress
+//       show progress + status
+//       else if
+//       recognizedText
+//       show AddDocumentForm
```
Comment (lmachens) [inline src/app/pages/Scan/Scan.tsx:63]:
```suggestion
```

File: src/app/Components/AddDocumentForm/AddDocumentForm.tsx (hunk 5 above)
Comment (lajanzen) [inline src/app/Components/AddDocumentForm/AddDocumentForm.tsx:40]:
```suggestion
      </div>
```

**Review by lajanzen — APPROVED** [review]

**Review by AnkePfei2fer — APPROVED** [review]
lgtm 🤖

**Pull request merged**